app/jobs/iiif_print/jobs/create_relationships_job.rb

```diff
@@ -103,25 +113,50 @@ def create_relationships(parent:, ordered_children:)
               @errors << e
             end
           end
-          parent.save! if @parent_record_members_added && @number_of_failures.zero?
+
+          if @parent_record_members_added && @number_of_failures.zero?
+            if parent.respond_to?(:save!)
+              parent.save!
+            else
+              Hyrax.persister.save(resource: parent)
+            end
+          end
         end
 
         # Bulkrax no longer reindexes file_sets, but IiifPrint needs both to add is_page_of_ssim for universal viewer.
         # This is where child works need to be indexed (AFTER the parent save), as opposed to how Bulkrax does it.
         ordered_children.each do |child_work|
-          child_work.update_index
-          child_work.file_sets.each(&:update_index) if child_work.respond_to?(:file_sets)
+          if child_work.respond_to?(:update_index)
```

Comment:
This switching logic could be extracted to the adapter logic:  

- lib/iiif_print/persistence_layer/active_fedora_adapter.rb
- lib/iiif_print/persistence_layer/valkyrie_adapter.rb

Create a class method `update_index(resource:)` and have the active fedora adapter implement the logic in the `if` portion; and the valkyrie_adapter implement the logic in the `else` portion.